### PR TITLE
Problem: no log converted from native event

### PR DESCRIPTION
**Pull request by mmsqe**
👮🏻👮🏻👮🏻 !!!! REFERENCE THE PROBLEM YOUR ARE SOLVING IN THE PR TITLE AND DESCRIBE YOUR SOLUTION HERE !!!! DO NOT FORGET !!!! 👮🏻👮🏻👮🏻

# PR Checklist:

- [ ] Have you read the [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md)?
- [ ] Does your PR follow the [C4 patch requirements](https://rfc.zeromq.org/spec:42/C4/#23-patch-requirements)?
- [ ] Have you rebased your work on top of the latest master? 
- [ ] Have you checked your code compiles? (`make`)
- [ ] Have you included tests for any non-trivial functionality?
- [ ] Have you checked your code passes the unit tests? (`make test`)
- [ ] Have you checked your code formatting is correct? (`go fmt`)
- [ ] Have you checked your basic code style is fine? (`golangci-lint run`)
- [ ] If you added any dependencies, have you checked they do not contain any known vulnerabilities? (`go list -json -m all | nancy sleuth`)
- [ ] If your changes affect the client infrastructure, have you run the integration test?
- [ ] If your changes affect public APIs, does your PR follow the [C4 evolution of public contracts](https://rfc.zeromq.org/spec:42/C4/#26-evolution-of-public-contracts)?
- [ ] If your code changes public APIs, have you incremented the crate version numbers and documented your changes in the [CHANGELOG.md](https://github.com/crypto-org-chain/chain-main/blob/master/CHANGELOG.md)?
- [ ] If you are contributing for the first time, please read the agreement in [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md) now and add a comment to this pull request stating that your PR is in accordance with the [Developer's Certificate of Origin](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md#developer-certificate-of-originn).

Thank you for your code, it's appreciated! :)

### CODE REVIEW

File: x/cronos/keeper/precompiles/base_contract.go [hunk 1]
```diff
@@ -1,27 +1,50 @@
...
+
+func (c *baseContract) ABIEvents() map[string]abi.Event {
+	return c.abi.Events
+}
+
+func (c *baseContract) CustomValueDecoders() ValueDecoders {
+	return nil
+}
+
+func (c *baseContract) PrecompileMethods() Methods {
```
Comment (yihuang) [inline x/cronos/keeper/precompiles/base_contract.go:48]: this method is not used.

File: x/cronos/events/event.go [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+	return results, nil
+}
+
+func (desc *EventDescriptor) ConvertEvent(
+	event []abci.EventAttribute,
+	valueDecoders ValueDecoders,
+) (*ethtypes.Log, error) {
+	attrs := make(map[string]string, len(event))
+	for _, attr := range event {
+		attrs[toMixedCase(attr.Key)] = attr.Value
```
[the remark this answers is not in the document]
Comment (mmsqe) [reply x/cronos/events/event.go:67]: toMixedCase is to convert like [client_id](https://github.com/cosmos/ibc-go/blob/main/modules/core/02-client/types/events.go#L11) to [clientId](https://github.com/crypto-org-chain/cronos/pull/1159/files#diff-42a43e4c9970ae7df0969fd1e252e5c37254849e1f34d43a1deed6ef46c8f0cbR13)

[the remark this answers is not in the document]
Comment (mmsqe) [reply x/cronos/events/event.go:67]: seems lost some customValueDecoders

File: x/cronos/events/events.go [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+func init() {
+	var ibcABI abi.ABI
+	if err := ibcABI.UnmarshalJSON([]byte(generated.RelayerModuleMetaData.ABI)); err != nil {
+		panic(err)
+	}
+	IBCEvents = NewEventDescriptors(ibcABI)
+}
+
+func ConvertEvent(event sdk.Event) (*ethtypes.Log, error) {
+	if event.Type == sdk.EventTypeMessage {
```
Comment (yihuang) [inline x/cronos/events/events.go:36]: Is this needed? if the `event.Type` is not found in `IBCEvents`, it'll be skipped anyway?

File: x/cronos/events/event.go [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+	for _, name := range attrNames {
+		value, ok := attrs[name]
+		if !ok {
+			return nil, fmt.Errorf("attribute %s not found", name)
+		}
+		decode, ok := valueDecoders[name]
+		if !ok {
+			return nil, fmt.Errorf("value decoder for %s not found", name)
+		}
+		values, err := decode(value, true)
```
Comment (yihuang) [inline x/cronos/events/event.go:52]: is the `boolean` parameter different for index/non-index?

File: x/cronos/events/events.go (hunk 3 above)
Comment (mmsqe) [reply x/cronos/events/events.go:36]: I see, since before fallback is returnStringAsIs if not found

File: x/cronos/events/event.go (hunk 4 above)
Comment (mmsqe) [reply x/cronos/events/event.go:52]: yea, true for topic, false for data

File: nix/gen-binding-shell.nix [hunk 5]
```diff
@@ -0,0 +1,10 @@
+{ system ? builtins.currentSystem, pkgs ? import ../nix { inherit system; } }:
+let
+  renameExe = pkgs.callPackage ../nix/rename-exe.nix { };
```
Comment (yihuang) [inline nix/gen-binding-shell.nix:3]:
```suggestion
  renameExe = pkgs.callPackage ./rename-exe.nix { };
```

File: nix/gen-binding-shell.nix [hunk 6]
```diff
@@ -0,0 +1,10 @@
+{ system ? builtins.currentSystem, pkgs ? import ../nix { inherit system; } }:
```
Comment (yihuang) [inline nix/gen-binding-shell.nix:1]:
```suggestion
{ system ? builtins.currentSystem, pkgs ? import . { inherit system; } }:
```

**Review by yihuang — APPROVED** [review]

**Pull request merged**